src/ontodia/diagram/paperArea.tsx

```diff
@@ -94,7 +95,25 @@ interface PointerMoveState {
     restoreGeometry: RestoreGeometry;
 }
 
+interface ViewportState {
+    readonly center: Vector; // paper coords
+    readonly scale: Vector;
+}
+
+interface ViewportAnimation {
+    readonly from: ViewportState;
+    readonly to: ViewportState;
+    readonly cancellation: Cancellation;
+}
+
+interface ViewPortAnimationDescription {
```

Comment:
I think we should get rid of this object and use `duration` directly to all animated methods (`duration === 0` means no animation is required).